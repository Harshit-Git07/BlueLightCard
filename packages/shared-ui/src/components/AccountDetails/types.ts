export type AccountDetailsProps = {
  accountNumber: string;
  firstName: string;
  lastName: string;
};
