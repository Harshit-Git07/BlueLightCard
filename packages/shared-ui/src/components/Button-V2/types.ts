import { MouseEventHandler, PropsWithChildren } from 'react';
import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { ThemeVariant } from '../../types';

export type ButtonProps = PropsWithChildren & {
  className?: string;
  type?: 'button' | 'submit';
  disabled?: boolean;
  slim?: boolean;
  variant?: ThemeVariant;
  invertColor?: boolean;
  iconLeft?: IconDefinition;
  iconRight?: IconDefinition;
  href?: string;
  newTab?: boolean;
  onClick?: MouseEventHandler;
  withoutHover?: boolean;
  withoutFocus?: boolean;
  borderless?: boolean;
  size?: ButtonSize;
  'data-testid'?: string;
};

export type ButtonSize = 'XSmall' | 'Small' | 'Large';
