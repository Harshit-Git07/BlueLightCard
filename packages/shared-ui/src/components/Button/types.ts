import { MouseEventHandler, PropsWithChildren } from 'react';
import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { ThemeVariant } from '../../types';

export type ColorToken = {
  bg: string;
  hover: string;
  focus: string;
  text: string;
  border: string;
};

export type ButtonProps = PropsWithChildren & {
  type?: 'button' | 'submit' | 'link';
  disabled?: boolean;
  slim?: boolean;
  className?: string;
  variant?: ThemeVariant;
  invertColor?: boolean;
  iconLeft?: IconDefinition;
  iconRight?: IconDefinition;
  href?: string;
  onClick?: MouseEventHandler;
  withoutHover?: boolean;
  withoutFocus?: boolean;
  borderless?: boolean;
};
