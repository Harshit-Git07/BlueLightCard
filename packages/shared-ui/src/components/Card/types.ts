import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { RefObject } from 'react';

export type CardState = 'default' | 'hover' | 'selected';

export interface CardProps {
  initialCardState?: CardState;
  cardTitle?: string;
  description?: string;
  truncateDescription?: boolean
  buttonTitle?: string;
  imageAlt?: string;
  imageSrc?: string;
  iconLeft?: IconDefinition;
  iconRight?: IconDefinition;
  showImage?: boolean;
  showDescription?: boolean;
  showButton?: boolean;
  onClick?: () => void;
  ariaLabel?: string;
  className?: string;
  'data-testid'?: string;
  canHover?: boolean;
}

export interface CardContentProps {
  title: string;
  description?: string;
  showDescription?: boolean;
  centerContent?: boolean;
  ariaLabel?: string;
  descriptionRef: RefObject<HTMLParagraphElement>;
  truncateDescription?: boolean;
}
