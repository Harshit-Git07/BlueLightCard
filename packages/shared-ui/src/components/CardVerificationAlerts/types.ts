import type { components } from '../../generated/MembersApi';

export type ProfileSchema = components['schemas']['ProfileModel'];
export type ApplicationSchema = ProfileSchema['applications'][number] & {
  documents?: string[];
  county?: string;
};

export type ApplicationReasonSchema = ApplicationSchema['applicationReason'];
export const allowedApplicationReasonValues: Array<ApplicationReasonSchema> = [
  'SIGNUP',
  'RENEWAL',
  'LOST_CARD',
  'REPRINT',
  'NAME_CHANGE',
];
export const allowedEligibilityStatusValues = [
  'ELIGIBLE',
  'AWAITING_ID_APPROVAL',
  'ASSIGNED_FOR_APPROVAL',
  'INELIGIBLE',
  undefined,
] as const;
export type EligibilityStatusSchema = (typeof allowedEligibilityStatusValues)[number];

export const allowedPaymentStatusValues = [
  'AWAITING_PAYMENT',
  'PAID_CARD',
  'PAID_PAYPAL',
  'PAID_PROMO_CODE',
  'PAID_CHEQUE',
  'PAID_ADMIN',
  'PENDING_REFUND',
  'REFUNDED',
  undefined,
] as const;
export type PaymentStatusSchema = (typeof allowedPaymentStatusValues)[number];

export const allowedVerificationStatusValues = ['ELIGIBLE', 'other', undefined] as const;
export type VerificationStatusSchema = (typeof allowedVerificationStatusValues)[number];

export type RejectionReasonSchema = ApplicationSchema['rejectionReason'];
export const allowedRejectionReasonValues: Array<RejectionReasonSchema> = [
  'BLURRY_IMAGE_DECLINE_ID',
  'DATE_DECLINE_ID',
  'DECLINE_INCORRECT_ID',
  'DIFFERENT_NAME_DECLINE_ID',
  'DECLINE_NOT_ELIGIBLE',
  'DL_PP_DECLINE_ID',
  'PASSWORD_PROTECTED_DECLINE_ID',
  'PICTURE_DECLINE_ID',
  undefined,
];
