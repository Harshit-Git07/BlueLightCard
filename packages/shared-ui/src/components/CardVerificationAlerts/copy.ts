export const copy = {
  awaitingId: {
    title: 'Get your {brandTitle}',
    subtext: 'Access exclusive offers for the next {cardDuration} for Â£4.99',
    buttonText: {
      awaitingId: 'Get your card',
      awaitingPayment: 'Make payment',
    },
  },
  awaitingIdApproval: {
    title: 'Your eligibility is being verified',
    subtext: 'You will receive a confirmation email, and your virtual card will be ready shortly',
    buttonText: 'Need help?',
  },
  rejected: {
    title: 'Your eligibility proof has been declined',
    subtext: {
      BLURRY_IMAGE_DECLINE_ID:
        'Your image is not clear enough. Provide a valid proof of eligibility.',
      DATE_DECLINE_ID: 'Your payslip is older than 3 months. Provide a valid proof of eligibility.',
      DECLINE_INCORRECT_ID: 'Your uploaded ID is invalid. Provide a valid proof of eligibility.',
      DIFFERENT_NAME_DECLINE_ID:
        'The name on your ID does not match your account. Provide a valid proof of eligibility.',
      DECLINE_NOT_ELIGIBLE:
        'Your proof of employment is invalid. Check your email resolve the issue.',
      DL_PP_DECLINE_ID:
        'You uploaded a driving license or passport. Provide a valid proof of eligibility.',
      PASSWORD_PROTECTED_DECLINE_ID:
        'Your ID is password-protected. Provide an accessible proof of eligibility.',
      PICTURE_DECLINE_ID: 'You uploaded a photo of yourself. Provide a valid proof of eligibility.',
    },
    buttonText: 'Verify your eligibility',
  },
} as const;
