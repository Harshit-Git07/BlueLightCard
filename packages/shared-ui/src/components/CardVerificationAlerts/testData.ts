import { ApplicationSchema, ProfileSchema } from './types';

export const defaultApplication: ApplicationSchema = {
  applicationId: '',
  applicationReason: 'SIGNUP',
  memberId: 'test_id',
};
export const defaultProfile: ProfileSchema = {
  cards: [],
  email: '',
  memberId: '',
  dateOfBirth: '',
  emailValidated: false,
  firstName: '',
  lastName: '',
  spareEmailValidated: false,
  applications: [],
};
