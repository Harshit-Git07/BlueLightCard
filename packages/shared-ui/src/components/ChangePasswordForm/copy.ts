const changePasswordFormCopy = {
  title: 'Change Password',
  primaryButtonLabel: 'Save',
  secondaryButtonLabel: 'Cancel',
  currentPasswordInput: {
    label: 'Current password',
    placeholder: 'Add current password',
  },
  newPasswordInput: {
    label: 'New password',
    placeholder: 'Add new password',
  },
  newPasswordConfirmInput: {
    label: 'Confirm new password',
    placeholder: 'Confirm new password',
  },
  validation: {
    missingCurrent: 'The current password is required.',
    incorrectCurrentPassword: 'Please enter a valid current password.',
    invalidPasswordRequirements: 'Please enter a valid password.',
    notNew: 'The new password needs to be different from current password.',
    doesNotMatch: 'Please enter a matching password.',
  },
};

export default changePasswordFormCopy;
