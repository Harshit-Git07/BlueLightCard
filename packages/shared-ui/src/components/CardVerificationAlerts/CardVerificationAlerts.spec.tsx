import { ApplicationSchema, ProfileSchema } from './types';
import { render, screen } from '@testing-library/react';
import CardVerificationBanner from '.';
import { copy } from './copy';
import { defaultApplication, defaultProfile } from './testData';
import { ApplicationReason } from '@blc-mono/shared/models/members/enums/ApplicationReason';
import { EligibilityStatus } from '@blc-mono/shared/models/members/enums/EligibilityStatus';
import { RejectionReason } from '@blc-mono/shared/models/members/enums/RejectionReason';

const mockUseMemberApplication = jest.fn();
jest.mock('../../hooks/useMemberApplication', () => ({
  __esModule: true,
  default: () => mockUseMemberApplication(),
}));

const awaitingIdProfile: ApplicationSchema = {
  ...defaultApplication,
  applicationReason: ApplicationReason.SIGNUP,
  eligibilityStatus: EligibilityStatus.INELIGIBLE,
};
const awaitingIdApprovalProfile: ApplicationSchema = {
  ...defaultApplication,
  applicationReason: ApplicationReason.SIGNUP,
  eligibilityStatus: EligibilityStatus.AWAITING_ID_APPROVAL,
  verificationMethod: 'other',
};
const rejectedProfile: ApplicationSchema = {
  ...defaultApplication,
  applicationReason: ApplicationReason.SIGNUP,
  eligibilityStatus: EligibilityStatus.INELIGIBLE,
  rejectionReason: RejectionReason.BLURRY_IMAGE_DECLINE_ID,
};

describe('CardVerificationAlert', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders awaitingId banner', () => {
    mockUseMemberApplication.mockReturnValue({ application: awaitingIdProfile });

    const { container } = render(<CardVerificationBanner />);

    expect(
      screen.getByText(copy.awaitingId.title.replace('{brandTitle}', 'Blue Light Card')),
    ).toBeInTheDocument();
    expect(screen.getByRole('link')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('renders awaitingIdApproval banner', () => {
    mockUseMemberApplication.mockReturnValue({ application: awaitingIdApprovalProfile });

    const { container } = render(<CardVerificationBanner />);

    expect(screen.getByText(copy.awaitingIdApproval.title)).toBeInTheDocument();
    expect(screen.getByRole('link')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('renders rejection banner', () => {
    mockUseMemberApplication.mockReturnValue({ application: rejectedProfile });

    const { container } = render(<CardVerificationBanner />);

    expect(screen.getByText(copy.rejected.title)).toBeInTheDocument();
    expect(screen.getByRole('link')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('renders nothing for any other application reason', () => {
    const otherApplicationReasonProfile: ProfileSchema = {
      ...defaultProfile,
      applications: [
        {
          ...defaultApplication,
          applicationReason: ApplicationReason.NAME_CHANGE,
        },
      ],
    };
    mockUseMemberApplication.mockReturnValue({ application: otherApplicationReasonProfile });

    const { container } = render(<CardVerificationBanner />);

    expect(container).toBeEmptyDOMElement();
  });

  it('renders nothing for no profile', () => {
    mockUseMemberApplication.mockReturnValue({ application: null });

    const { container } = render(<CardVerificationBanner />);

    expect(container).toBeEmptyDOMElement();
  });

  it('renders nothing for no applications', () => {
    mockUseMemberApplication.mockReturnValue({ data: {} });

    const { container } = render(<CardVerificationBanner />);

    expect(container).toBeEmptyDOMElement();
  });
});
