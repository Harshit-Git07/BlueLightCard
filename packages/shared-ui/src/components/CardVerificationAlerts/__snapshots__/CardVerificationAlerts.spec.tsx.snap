// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardVerificationAlert renders awaitingId banner 1`] = `
<div>
  <div
    class="
         
        w-full
        bg-colour-warning-bright-light dark:bg-colour-warning-bright-dark 
        sticky top-0 left-0 z-50 px-[1rem] tablet:px-[3.5rem] laptop:px-[6.5rem] py-[1rem] 
        flex flex-col rounded justify-between 
      "
    data-testid="alert"
  >
    <div
      class="flex basis-full flex-row"
      data-testid="alertBanner"
    >
      <div
        class="flex items-start tablet:items-center"
      >
        <div
          class="text-colour-onWarning-container-light dark:text-colour-onWarning-container-dark"
          data-testid="alert-icon"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-triangle-exclamation text-[20px] flex content-start pt-1 tablet:pt-[0px] tablet:text-[24px]"
            data-icon="triangle-exclamation"
            data-prefix="fas"
            data-testid="icon"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 32c14.2 0 27.3 7.5 34.5 19.8l216 368c7.3 12.4 7.3 27.7 .2 40.1S486.3 480 472 480H40c-14.3 0-27.6-7.7-34.7-20.1s-7-27.8 .2-40.1l216-368C228.7 39.5 241.8 32 256 32zm0 128c-13.3 0-24 10.7-24 24V296c0 13.3 10.7 24 24 24s24-10.7 24-24V184c0-13.3-10.7-24-24-24zm32 224a32 32 0 1 0 -64 0 32 32 0 1 0 64 0z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
      <div
        class="tablet:flex tablet:mx-6 ml-4 m-auto w-full"
      >
        <div
          class="flex-col content-center  max-w-[750px] tablet:pr-[16px]"
        >
          <p
            class="w-full text-colour-onSurface-light dark:text-colour-onSurface-dark break-words font-typography-title-small font-typography-title-small-weight text-typography-title-small leading-typography-title-small tracking-typography-title-small tablet:font-typography-title-medium-semibold tablet:font-typography-title-medium-semibold-weight tablet:text-typography-title-medium-semibold tablet:leading-typography-title-medium-semibold tablet:tracking-typography-title-medium-semibold"
          >
            Get your Blue Light Card
          </p>
          <div
            class="flex-col w-full pt-2 font-typography-body-small font-typography-body-small-weight text-typography-body-small leading-typography-body-small tracking-typography-body-small tablet:font-typography-body tablet:font-typography-body-weight tablet:text-typography-body tablet:leading-typography-body tablet:tracking-typography-body text-colour-onSurface-subtle-light dark:text-colour-onSurface-subtle-dark break-words"
          >
            Access exclusive offers for the next 2 years for Â£4.99
          </div>
        </div>
        <div
          class="mt-[0.75rem] tablet:mt-0 tablet:justify-end flex-row ml-auto mr-4 items-center flex justify-start items-start max-w-[250px] text-base w-full"
        >
          <div
            class="h-full flex items-center tablet:ml-auto"
          >
            <a
              class="rounded transition border-1 py-[4px] px-[12px] h-[32px] flex items-center justify-center gap-[8px]  text-button-primary-default-label-colour-light dark:text-button-primary-default-label-colour-dark hover:bg-button-primary-hover-bg-colour-light dark:hover:bg-button-primary-hover-bg-colour-dark hover:text-button-primary-hover-label-colour-light dark:hover:text-button-primary-hover-label-colour-dark focus:outline-[#2EB8E6] dark:focus:outline-[#FFFF00] bg-button-primary-default-bg-colour-light dark:bg-button-primary-default-bg-colour-dark border-none  text-button-label-font font-button-label-font font-button-label-font-weight tracking-button-label-font leading-button-label-font"
              href="/eligibility"
            >
              Get your card
            </a>
          </div>
        </div>
      </div>
      <div
        class="flex items-start  tablet:items-center"
      >
        <button
          aria-label="Close"
          class="content-center text-center rounded-full text-colour-onSurface-subtle-light dark:text-colour-onSurface-subtle-dark"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-xmark text-[20px]"
            data-icon="xmark"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M345 137c9.4-9.4 9.4-24.6 0-33.9s-24.6-9.4-33.9 0l-119 119L73 103c-9.4-9.4-24.6-9.4-33.9 0s-9.4 24.6 0 33.9l119 119L39 375c-9.4 9.4-9.4 24.6 0 33.9s24.6 9.4 33.9 0l119-119L311 409c9.4 9.4 24.6 9.4 33.9 0s9.4-24.6 0-33.9l-119-119L345 137z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`CardVerificationAlert renders awaitingIdApproval banner 1`] = `
<div>
  <div
    class="
         
        w-full
        bg-colour-info-bright-light dark:bg-colour-info-bright-dark 
        sticky top-0 left-0 z-50 px-[1rem] tablet:px-[3.5rem] laptop:px-[6.5rem] py-[1rem] 
        flex flex-col rounded justify-between 
      "
    data-testid="alert"
  >
    <div
      class="flex basis-full flex-row"
      data-testid="alertBanner"
    >
      <div
        class="flex items-start tablet:items-center"
      >
        <div
          class="text-colour-onInfo-container-light dark:text-colour-onInfo-container-dark"
          data-testid="alert-icon"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-info text-[20px] flex content-start pt-1 tablet:pt-[0px] tablet:text-[24px]"
            data-icon="circle-info"
            data-prefix="fas"
            data-testid="icon"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM216 336h24V272H216c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24H216c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-208a32 32 0 1 1 0 64 32 32 0 1 1 0-64z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
      <div
        class="tablet:flex tablet:mx-6 ml-4 m-auto w-full"
      >
        <div
          class="flex-col content-center  max-w-[750px] tablet:pr-[16px]"
        >
          <p
            class="w-full text-colour-onSurface-light dark:text-colour-onSurface-dark break-words font-typography-title-small font-typography-title-small-weight text-typography-title-small leading-typography-title-small tracking-typography-title-small tablet:font-typography-title-medium-semibold tablet:font-typography-title-medium-semibold-weight tablet:text-typography-title-medium-semibold tablet:leading-typography-title-medium-semibold tablet:tracking-typography-title-medium-semibold"
          >
            Your eligibility is being verified
          </p>
          <div
            class="flex-col w-full pt-2 font-typography-body-small font-typography-body-small-weight text-typography-body-small leading-typography-body-small tracking-typography-body-small tablet:font-typography-body tablet:font-typography-body-weight tablet:text-typography-body tablet:leading-typography-body tablet:tracking-typography-body text-colour-onSurface-subtle-light dark:text-colour-onSurface-subtle-dark break-words"
          >
            You will receive a confirmation email, and your virtual card will be ready shortly
          </div>
        </div>
        <div
          class="mt-[0.75rem] tablet:mt-0 tablet:justify-end flex-row ml-auto mr-4 items-center flex justify-start items-start max-w-[250px] text-base w-full"
        >
          <div
            class="h-full flex items-center tablet:ml-auto"
          >
            <a
              class="rounded transition border-1 h-[32px] flex items-center justify-center gap-[8px]  text-button-tertiary-default-label-colour-light dark:text-button-tertiary-default-label-colour-dark hover:text-button-tertiary-hover-label-colour-light dark:hover:text-button-tertiary-hover-label-colour-dark focus:outline-[#2EB8E6] dark:focus:outline-[#FFFF00]  border-none !px-0 text-button-label-font font-button-label-font font-button-label-font-weight tracking-button-label-font leading-button-label-font"
              href="https://support.bluelightcard.co.uk/hc/en-gb/requests/new"
            >
              Need help?
            </a>
          </div>
        </div>
      </div>
      <div
        class="flex items-start  tablet:items-center"
      >
        <button
          aria-label="Close"
          class="content-center text-center rounded-full text-colour-onSurface-subtle-light dark:text-colour-onSurface-subtle-dark"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-xmark text-[20px]"
            data-icon="xmark"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M345 137c9.4-9.4 9.4-24.6 0-33.9s-24.6-9.4-33.9 0l-119 119L73 103c-9.4-9.4-24.6-9.4-33.9 0s-9.4 24.6 0 33.9l119 119L39 375c-9.4 9.4-9.4 24.6 0 33.9s24.6 9.4 33.9 0l119-119L311 409c9.4 9.4 24.6 9.4 33.9 0s9.4-24.6 0-33.9l-119-119L345 137z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`CardVerificationAlert renders rejection banner 1`] = `
<div>
  <div
    class="
         
        w-full
        bg-colour-warning-bright-light dark:bg-colour-warning-bright-dark 
        sticky top-0 left-0 z-50 px-[1rem] tablet:px-[3.5rem] laptop:px-[6.5rem] py-[1rem] 
        flex flex-col rounded justify-between 
      "
    data-testid="alert"
  >
    <div
      class="flex basis-full flex-row"
      data-testid="alertBanner"
    >
      <div
        class="flex items-start tablet:items-center"
      >
        <div
          class="text-colour-onWarning-container-light dark:text-colour-onWarning-container-dark"
          data-testid="alert-icon"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-triangle-exclamation text-[20px] flex content-start pt-1 tablet:pt-[0px] tablet:text-[24px]"
            data-icon="triangle-exclamation"
            data-prefix="fas"
            data-testid="icon"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 32c14.2 0 27.3 7.5 34.5 19.8l216 368c7.3 12.4 7.3 27.7 .2 40.1S486.3 480 472 480H40c-14.3 0-27.6-7.7-34.7-20.1s-7-27.8 .2-40.1l216-368C228.7 39.5 241.8 32 256 32zm0 128c-13.3 0-24 10.7-24 24V296c0 13.3 10.7 24 24 24s24-10.7 24-24V184c0-13.3-10.7-24-24-24zm32 224a32 32 0 1 0 -64 0 32 32 0 1 0 64 0z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
      <div
        class="tablet:flex tablet:mx-6 ml-4 m-auto w-full"
      >
        <div
          class="flex-col content-center  max-w-[750px] tablet:pr-[16px]"
        >
          <p
            class="w-full text-colour-onSurface-light dark:text-colour-onSurface-dark break-words font-typography-title-small font-typography-title-small-weight text-typography-title-small leading-typography-title-small tracking-typography-title-small tablet:font-typography-title-medium-semibold tablet:font-typography-title-medium-semibold-weight tablet:text-typography-title-medium-semibold tablet:leading-typography-title-medium-semibold tablet:tracking-typography-title-medium-semibold"
          >
            Your eligibility proof has been declined
          </p>
          <div
            class="flex-col w-full pt-2 font-typography-body-small font-typography-body-small-weight text-typography-body-small leading-typography-body-small tracking-typography-body-small tablet:font-typography-body tablet:font-typography-body-weight tablet:text-typography-body tablet:leading-typography-body tablet:tracking-typography-body text-colour-onSurface-subtle-light dark:text-colour-onSurface-subtle-dark break-words"
          >
            Your image is not clear enough. Provide a valid proof of eligibility.
          </div>
        </div>
        <div
          class="mt-[0.75rem] tablet:mt-0 tablet:justify-end flex-row ml-auto mr-4 items-center flex justify-start items-start max-w-[250px] text-base w-full"
        >
          <div
            class="h-full flex items-center tablet:ml-auto"
          >
            <a
              class="rounded transition border-1 py-[4px] px-[12px] h-[32px] flex items-center justify-center gap-[8px]  text-button-primary-default-label-colour-light dark:text-button-primary-default-label-colour-dark hover:bg-button-primary-hover-bg-colour-light dark:hover:bg-button-primary-hover-bg-colour-dark hover:text-button-primary-hover-label-colour-light dark:hover:text-button-primary-hover-label-colour-dark focus:outline-[#2EB8E6] dark:focus:outline-[#FFFF00] bg-button-primary-default-bg-colour-light dark:bg-button-primary-default-bg-colour-dark border-none  text-button-label-font font-button-label-font font-button-label-font-weight tracking-button-label-font leading-button-label-font"
              href="/eligibility"
            >
              Verify your eligibility
            </a>
          </div>
        </div>
      </div>
      <div
        class="flex items-start  tablet:items-center"
      >
        <button
          aria-label="Close"
          class="content-center text-center rounded-full text-colour-onSurface-subtle-light dark:text-colour-onSurface-subtle-dark"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-xmark text-[20px]"
            data-icon="xmark"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M345 137c9.4-9.4 9.4-24.6 0-33.9s-24.6-9.4-33.9 0l-119 119L73 103c-9.4-9.4-24.6-9.4-33.9 0s-9.4 24.6 0 33.9l119 119L39 375c-9.4 9.4-9.4 24.6 0 33.9s24.6 9.4 33.9 0l119-119L311 409c9.4 9.4 24.6 9.4 33.9 0s9.4-24.6 0-33.9l-119-119L345 137z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
